pkg/cache/interface.go

```diff
@@ -0,0 +1,12 @@
+package cache
+
+// Interface defines a generic concurrency safe key-value cache used to
+// communicate between packages.
+type Interface[K comparable, V any] interface {
```

Comment:
This interface makes it easier to reason about the purpose of the cache channels and simplifies their usage by way of standardizing access patterns, regardless of the underlying implementation details.